src/compiler/checker/lines_32000_40000.rs

```diff
@@ -193,10 +188,7 @@ impl TypeChecker {
         check_mode: Option<CheckMode>,
     ) -> Rc<Type> {
         self.check_expression_for_mutable_location(
-            match &*node.initializer {
-                Node::Expression(expression) => expression,
-                _ => panic!("Expected Expression"),
-            },
+            enum_unwrapped!(&*node.initializer, [Node, Expression]),
```

Comment:
Another possible convenience for unwrapping is where eg a `Node` variant has an `Rc<Node>` field that's of a known sub-variant (eg here we know that `PropertyAssignment.initializer` is always an `Expression`), expose a custom eg `.initializer_as_expression()` helper that does the unwrapping for you (although would lifetimes get weird if that tried to return a reference when called against an `Rc<Node>` variable?)